legend-engine-xt-data-space-generation/src/main/java/org/finos/legend/engine/generation/DataSpaceAnalyticsArtifactGenerationExtension.java

```diff
@@ -63,15 +62,12 @@ public List<Artifact> generate(PackageableElement element, PureModel pureModel,
         try
         {
             String stringResult = objectMapper.writeValueAsString(result);
-            String filePath = Root_meta_pure_functions_meta_elementToPath_PackageableElement_1__String_1__String_1_(element, "/", pureModel.getExecutionSupport()) + ".json";
-            Artifact output = new Artifact(stringResult, filePath, "json");
+            Artifact output = new Artifact(stringResult, fileName, "json");
             return Collections.singletonList(output);
-
         }
         catch (Exception ignored)
         {
-            // ignore
+            throw new RuntimeException("Error serializing dataSpace analysis result for dataSpace '" + dataSpacePath + "'");
```

Comment:
Please throw an EngineException with an appropriate EngineErrorType.

https://github.com/finos/legend-engine/blob/master/legend-engine-shared-core/src/main/java/org/finos/legend/engine/shared/core/operational/errorManagement/EngineException.java

Right now we throw RuntimeExceptions all over the place which makes it harder to categorize exceptions for troubleshooting.

Reply:
EngineException is better than RuntimeException. Can we also include an appropriate EngineErrorType / ExceptionCategory ?

Reply:
There is no existing ErrorType that makes sense here. 
The available ones are COMPLIATION and PARSER. 
I believe this eninge error exceptions are due to be refactored so adding new error types also doesn't make sense.